apps/molgenis-components/src/components/tables/TableExplorer.vue

```diff
@@ -580,99 +607,47 @@ export default {
       this.loading = false;
     },
   },
-  watch: {
-    searchTerms: {
-      handler(newValue) {
-        this.$emit("searchTerms", newValue);
-        this.reload();
-      },
-    },
-    page() {
-      this.loading = true;
-      this.offset = this.limit * (this.page - 1);
-      this.$emit("update:showPage", this.page);
-      this.reload();
-    },
-    showOrderBy() {
-      this.orderByColumn = this.showOrderBy;
-      this.$emit("update:showOrderBy", this.showOrderBy);
-    },
-    showOrder() {
-      this.order = this.showOrder;
-      this.$emit("update:showOrder", this.showOrder);
-      this.reload();
-    },
-    showPage() {
-      this.page = this.showPage;
-    },
-    showLimit() {
-      this.limit = this.showLimit;
-    },
-    tableMetadata() {
-      this.page = this.showPage;
-      this.limit = this.showLimit;
-      this.orderByColumn = this.showOrderBy;
-      this.order = this.showOrder;
-      if (this.columns.length === 0) {
-        this.columns.push(...this.tableMetadata.columns);
-        // //init settings
-        this.columns.forEach((c) => {
-          //show columns
-          if (this.showColumns && this.showColumns.length > 0) {
-            if (this.showColumns.includes(c.name)) {
-              c.showColumn = true;
-            } else {
-              c.showColumn = false;
-            }
-          } else {
-            //default we show all non mg_ columns
-            if (!c.name.startsWith("mg_")) {
-              c.showColumn = true;
-            } else {
-              c.showColumn = false;
-            }
-          }
-          //show filters
-          if (this.showFilters && this.showFilters.length > 0) {
-            if (this.showFilters.includes(c.name)) {
-              c.showFilter = true;
-            } else {
-              c.showFilter = false;
-            }
-          } else {
-            //default we hide all filters
-            c.showFilter = false;
-          }
-        });
-        if (this.showView) {
-          this.view = this.showView;
-        }
-        this.columns.forEach((c) => {
-          if (this.conditions[c.name] && this.conditions[c.name].length > 0) {
-            this.$set(c, "conditions", this.conditions[c.name]); //haat vue reactivity
-          } else {
-            c.conditions = [];
-          }
-        });
-        //table settings
-        if (this.tableMetadata.settings) {
-          this.tableMetadata.settings.forEach((s) => {
-            if (s.key == "cardTemplate") this.cardTemplate = s.value;
-            if (s.key == "recordTemplate") this.recordTemplate = s.value;
-          });
-        }
-      }
-      this.reload();
-    },
-  },
   mounted: async function () {
+    this.client = Client.newClient(this.graphqlURL);
+    const newTableMetadata = await this.client
```

Comment:
could metadata change after mount ? , maybe not

Reply:
It should not be able to.

Reply:
al the features seem to work without changing the metadata